### PR TITLE
`azurerm_role_assignment`: normalize `scope` field in Read

### LINKED ISSUE #23487: `azurerm_role_assignment` forces replace after import
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.4.6

### AzureRM Provider Version

3.55.0

### Affected Resource(s)/Data Source(s)

azurerm_role_assignment

### Terraform Configuration Files

```hcl
resource "azurerm_role_assignment" "test" {
  scope                = data.azurerm_resource_group.rg-fs.id
  role_definition_name = "User Access Administrator"
  principal_id         = data.azuread_group.common-resourcecontributor.object_id
}

data "azuread_group" "common-resourcecontributor" { display_name = "group_name" }

data "azurerm_resource_group" "rg-fs" { name = "rg_name" }
```

### Debug Output/Panic Output

```shell
Terraform will perform the following actions:

  # azurerm_role_assignment.test must be replaced
-/+ resource "azurerm_role_assignment" "test" {
      ~ id                               = "/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Authorization/roleAssignments/REDACTED" -> (known after apply)
      ~ name                             = "REDACTED" -> (known after apply)
      ~ principal_type                   = "Group" -> (known after apply)
      ~ role_definition_id               = "/subscriptions/REDACTED/providers/Microsoft.Authorization/roleDefinitions/REDACTED" -> (known after apply)
      ~ scope                            = "/subscriptions/REDACTED/resourcegroups/REDACTED" -> "/subscriptions/REDACTED/resourceGroups/REDACTED" # forces replacement
      + skip_service_principal_aad_check = (known after apply)
        # (2 unchanged attributes hidden)

      - timeouts {}
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

### Expected Behaviour

`azurerm_role_assignment` should not be replaced due to forced lowercase of scope. This line:

```      
~ scope                            = "/subscriptions/REDACTED/resourcegroups/REDACTED" -> "/subscriptions/REDACTED/resourceGroups/REDACTED" # forces replacement
```

shouldn't be happening because the string imported contained the substring `resourceGroups` but it got lowercased by the provider during import

### Actual Behaviour

The provider detects there is a difference in casing between the imported scope and the reported one and attempts to do a replacement, because the scope got lowercased during import.

### Steps to Reproduce

1. `terraform import azurerm_role_assignment.test INSERT_CORRECTLY_CASED_ID_HERE`
2. `terraform apply`

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by wuxu92**
fixes #23487.

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @wuxu92 thanks for submitting this. I ran the tests and it looks like there is one failure:

```
------- Stdout: -------
=== RUN   TestAccRoleAssignment_ServicePrincipalGroup
=== PAUSE TestAccRoleAssignment_ServicePrincipalGroup
=== CONT  TestAccRoleAssignment_ServicePrincipalGroup
    testcase.go:113: Step 1/1 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
        stdout
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          + create
        -/+ destroy and then create replacement
        Terraform will perform the following actions:
          # azuread_group.test will be created
          + resource "azuread_group" "test" {
              + auto_subscribe_new_members     = (known after apply)
              + display_name                   = "acctestspa-231026150514623265"
              + external_senders_allowed       = (known after apply)
              + hide_from_address_lists        = (known after apply)
              + hide_from_outlook_clients      = (known after apply)
              + id                             = (known after apply)
              + mail                           = (known after apply)
              + mail_nickname                  = (known after apply)
              + members                        = (known after apply)
              + object_id                      = (known after apply)
              + onpremises_domain_name         = (known after apply)
              + onpremises_group_type          = (known after apply)
              + onpremises_netbios_name        = (known after apply)
              + onpremises_sam_account_name    = (known after apply)
              + onpremises_security_identifier = (known after apply)
              + onpremises_sync_enabled        = (known after apply)
              + owners                         = (known after apply)
              + preferred_language             = (known after apply)
              + prevent_duplicate_names        = false
              + proxy_addresses                = (known after apply)
              + security_enabled               = true
              + visibility                     = (known after apply)
              + writeback_enabled              = false
            }
          # azurerm_role_assignment.test must be replaced
        -/+ resource "azurerm_role_assignment" "test" {
              ~ id                               = "/subscriptions/*******/providers/Microsoft.Authorization/roleAssignments/7d8e80d0-8b03-4439-8e13-81f590323bec" -> (known after apply)
                name                             = "7d8e80d0-8b03-4439-8e13-81f590323bec"
              ~ principal_id                     = "9486cb64-853a-4d74-8316-3c9ce51010d8" # forces replacement -> (known after apply) # forces replacement
              ~ principal_type                   = "Group" -> (known after apply)
              ~ role_definition_id               = "/subscriptions/*******/providers/Microsoft.Authorization/roleDefinitions/acdd72a7-3385-48ef-bd42-f606fba81ae7" -> (known after apply)
              + skip_service_principal_aad_check = (known after apply)
                # (2 unchanged attributes hidden)
            }
        Plan: 2 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccRoleAssignment_ServicePrincipalGroup (239.08s)
FAIL
```

**Comment by wuxu92** [comment]
Hi @catriona-m , I reran the tests and they passed.

image

File: internal/services/authorization/role_assignment_resource.go [hunk 1]
```diff
@@ -413,3 +414,20 @@ func getTenantIdBySubscriptionId(ctx context.Context, client *subscriptions.Clie
...
+		return rg.ID()
+	}
+
+	if sub, err := commonids.ParseSubscriptionIDInsensitively(scope); err == nil {
+		return sub.ID()
+	}
+
+	if group, err := commonids.ParseManagementGroupIDInsensitively(scope); err == nil {
+		return group.ID()
+	}
```
Comment (catriona-m) [inline internal/services/authorization/role_assignment_resource.go:429]: we should only need to normalize the resourcegroup here

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks @wuxu92 - I left one comment inline but otherwise this is looking good! Thanks!

File: internal/services/authorization/role_assignment_resource.go (hunk 1 above)
Comment (wuxu92) [reply internal/services/authorization/role_assignment_resource.go]: Updated.

**Review by catriona-m — APPROVED** [review]
Thanks @wuxu92 LGTM!

**Pull request merged**